Google-Summer-of-Code-2023.md

```diff
@@ -273,3 +273,504 @@ The following is a list of Oppia's 2023 GSoC project ideas. You are welcome to c
+
+* Knowledge and understanding of Python
+* Knowledge and understanding of TypeScript, Angular, HTML, and CSS
+
+**Suggested Milestones:**
+
+* **Milestone 1:**
+  - Showing translation cards according to the order of the content flow in the lesson. Exploration editor should also be linked so that the original lesson can be seen.
+  - Showing the size of the translation(short/large)
+  - Pin/unpin specific lessons to translate
+
+
+* **Milestone 2:**
+  - Ability to revert the acceptance or rejection of cards. The exact time limit that we allow to undo the review should be decided.
+  - Showing image alt text below the image
+  - Notifying new submissions through emails
+
+
+**Dependency on Release Schedule:** 
+
+**Proposal notes:**
+
+* To support lesson pinning, we have to make sure the pinned lessons are returned at the top of the list of lessons with translation suggestions in the fetch query [here](https://github.com/oppia/oppia/blob/develop/core/controllers/contributor_dashboard.py#L336).
+* We will likely want to introduce a new storage model to track pinned lessons by language and user.
+
+
+**Useful resources:**
+
+## Server data validation team
+
+### 3.1. Fix Validation Errors
+
+TBA
+
+## Developer workflow team
+
+### 4.1. Dockerize Oppia
+
+**Project Description:**
+
+We want to package Oppia as a Docker container so that new contributors can get started by running docker-compose up -d after cloning the repository to install all the dependencies and start the local development server. This will be much easier than the current installation process, which is often painful to troubleshoot.
+
+Requirements:
+* Setup should be quick and easy. For example, a user might need to:
+  - Install Docker 
+  - Clone oppia/oppia 
+  - Run docker-compose up -d
+* The setup should be independent of the OS or platform it is run on.
+* The setup should not exceed the 8GB RAM requirement for Oppia on any platform for which we have supported installation instructions.
+* It should be easy for a user to update their installation to the current dependency versions.
+* It should be easy to reset the environment, for example if the internet connection fails and so a dependency is only partly installed.
+* Developers should be able to run tests, use git (e.g. switch branches, push changes, and checkout commits), and use all the other development tools we document in the wiki. The interface for doing so should undergo only minimal changes.
+* Maintainers should be able to easily upgrade Oppia’s dependencies (e.g. change requirements.in to specify updated versions) and check that the local server still works. When those updates are merged into develop, all developers should be smoothly upgraded to the new versions.
+* All dependencies must be pinned to a specific version and come from a trustworthy source. We should also verify dependency checksums before installing them.
+* All installations should happen via Docker or bash commands.
```

Comment:
```suggestion
* All installations should happen via Docker or standard command-line tools.
```

I'm discussing what the exact requirement is with @gp201, but I think this is closer to what we'll decide on